### PR TITLE
Smart linking in web apps: Stack queries

### PR DESCRIPTION
**Pull request by orangejenny**
Needed for https://github.com/dimagi/formplayer/pull/1014 but can be merged independently. This adds parsing for stack queries and has logic to turn a query into a `StackFrameStep`.

Also PRed against master: https://github.com/dimagi/commcare-core/pull/1056

Review full diff at once.

### CODE REVIEW

File: src/main/java/org/commcare/xml/StackFrameStepParser.java [hunk 1]
```diff
@@ -22,17 +23,41 @@
...
+                return parseValue(SessionFrame.STATE_UNKNOWN, datumId);
             case "rewind":
                 return parseValue(SessionFrame.STATE_REWIND, null);
             case "mark":
                 return parseValue(SessionFrame.STATE_MARK, null);
             case "command":
                 return parseValue(SessionFrame.STATE_COMMAND_ID, null);
+            case "query":
+                String queryId = parser.getAttributeValue(null, "id");
+                String url = parser.getAttributeValue(null, "value");
```
Comment (shubham1g5) [inline src/main/java/org/commcare/xml/StackFrameStepParser.java:40]: we should probably do some validation like [here](https://github.com/dimagi/commcare-core/blob/master/src/main/java/org/commcare/xml/SessionDatumParser.java#L98) on url

**Review by shubham1g5 — APPROVED** [review]
Nothing blocking but probably be good to add some tests around this probably[here](https://github.com/dimagi/commcare-core/blob/master/src/test/java/org/commcare/backend/session/test/SessionStackTests.java) just to test the parsing.

**Review by snopoke — COMMENTED** [review]
+1 on @shubham1g5 comments. Also this should get a docs update.

**Review by ctsims — APPROVED** [review]
Agreed with Simon that we should make sure that docs are updated.

Specifically, since this is very similar to another construct elsewhere in the app, but with slightly different semantics, it would be useful for someone to be able to figure out what expectations to have about the query lifecycle here (might have more feedback about that in the formplayer half of the PR.

File: src/main/java/org/commcare/xml/StackFrameStepParser.java (hunk 1 above)
Comment (orangejenny) [reply src/main/java/org/commcare/xml/StackFrameStepParser.java]: Simon addressed in https://github.com/dimagi/commcare-core/pull/1048/commits/d1ea42af5e893cb33a097583fd40c21b89027ade

**Comment by orangejenny** [comment]
@shubham1g5 What do you think of https://github.com/dimagi/commcare-core/pull/1048/commits/4ba1de63f0e53347257ab0e95293e1534dba8ef0 for tests? It doesn't really assert anything, but it it to execute the parsing code.

**Review by snopoke — APPROVED** [review]

**Review by shubham1g5 — APPROVED** [review]
yes, looks great!

**Comment by orangejenny** [comment]
Updated docs: https://github.com/dimagi/commcare-core/wiki/SessionStack#jump

**Pull request merged**